### DISCUSSION TITLE
Para descontrair: libs de dev preguiçoso

### DISCUSSION THREAD

**Discussion by kvnol** [frontendbr/forum#1303] [Perguntas e Respostas]
Vocês já pegaram algum projeto para dar manutenção e quando abriram o código viram uma coisa grotesca? Aquele trabalho porco que você olha e fala "PRA QUE ESSA LIB, MEU DEUS?"

Para começar, eu acabei de descobrir uma lib que nomeia automaticamente a metodologia BEM no HTML. Não vou divulgar o autor, pois não achei no GitHub, mas é basicamente isso: https://gist.github.com/kvnol/05b0fa0ea2c2311821112e5475a47573

Qual a lib mais sem sentido que vocês já viram em algum projeto?

PS: não fique ofendido caso alguém faça zoeira com a lib que você usa.
PS2: ok, sabemos que muitas dessas libs no fundo tem alguma utilidade real.

**Comment by waldemirflj** [comment]
Sempre, hoje mesmo peguei um projeto que o cara instalou o Lodash para fazer um filter/sort. 
A "senioridade" vem com o tempo e até lá olhos sangraram. Rs 

P.S.: Caso encontre o autor dessa lib questione o motivo da criação, e se tiver uma solução mais elegante mostre para ele.

**Comment by kvnol** [comment]
> P.S.: Caso encontre o autor dessa lib questione o motivo da criação, e se tiver uma solução mais elegante mostre para ele.

O autor trabalhava onde eu estou agora, o dev que está trabalhando comigo disse que ele só não queria ficar reescrevendo as classes.

**Comment by waldemirflj** [comment]
> > P.S.: Caso encontre o autor dessa lib questione o motivo da criação, e se tiver uma solução mais elegante mostre para ele.
> 
> O autor trabalhava onde eu estou agora, o dev que está trabalhando comigo disse que ele só não queria ficar reescrevendo as classes.

Rs...

**Comment by fabiomirandaa** [comment]
Antigamente nego botava JQuery nos projetos simples pra fazer coisas que o Vanilla já atendia de boas, como selecionar classes e etc hahaha

**Comment by vNNi** [comment]
Hoje em dia eu acho bizarro usar lib pra fazer http request, depois da Fetch API, porém entendo a necessidade, por não ser ainda suportada por 100% dos browsers

**Comment by ViniciusGularte** [comment]
Ja vi colocarem a lib inteira do font awesome so pra colocar um icone na pagina :laughing:

**Comment by ninetails** [comment]
Já fiz tantos na vida...

Mas por favor, não são "libs de dev preguiçoso", use a nomenclatura certa: **automatização de tarefas triviais**. :laughing:

**Comment by lagden** [comment]
😵 Se o dev é preguiçoso ele nem escreve lib!!

**Comment by klarkc** [comment]
A regra é clara, se existe uma lib, ela é incorporada

**Comment by didiraja** [comment]
> Ja vi colocarem a lib inteira do font awesome so pra colocar um icone na pagina 😆

@ViniciusGularte foi o que mais fiz e ficava agoniado sempre hahah

**Comment by nandomoreirame** [comment]
É comum devs fazerem códigos só por brincadeira, para estudar ou para automatizar uma tarefa mesmo, o motivo real o dev não precisa ficar explicando, ele simplesmente fez pq quis ou pq sabe fazer hahahahaha

Eu mesmo já parei pra fazer umas coisa aleatórias aqui só "porque sim" as vezes nem precisa de motivo haeuhauehue 😆

**Comment by nandomoreirame** [comment]
Quem nunca quis criar o seu próprio framework ou seu próprio jQuery só por diversão ou estudo que atire a primeira pedra 😆

**Comment by expalmer** [comment]
Que me diz de criar um `virtual dom` de pobre ?
[https://github.com/expalmer/tiny-virtual-dom](https://github.com/expalmer/tiny-virtual-dom)

**Comment by klarkc** [comment]
@expalmer já que mencionou criar o seu próprio virtual dom, https://github.com/danistefanovic/build-your-own-x (adoro o github)

**Comment by ViniciusGularte** [comment]
@klarkc  não sabia desse repo, github é vida :heart: 
@didiraja fiz muito, ficava com peso na consciência kkk

**Comment by valterbarros** [comment]
Não é bem lib de preguiçoso, mas olha o nome dessas duas libs aqui, meu palpite é que raphael e morris são primos ai o raphael fez a lib de chart e o morris fez a lib de tooltip, negocios em familia. kkkkk. Encontrei isso na empresa em que trabalho código legado é osso, ainda bem que to numa task para remover essas duas.

```
import "raphael/raphael"
import "morris.js/morris.min"
```
